modules/tinymce/src/core/main/ts/paste/Clipboard.ts

```diff
@@ -34,6 +35,9 @@ const doPaste = (editor: Editor, content: string, internal: boolean, pasteAsText
   if (!args.cancelled) {
     const content = args.content;
     SmartPaste.insertContent(editor, content, pasteAsText);
+    const dataTransfer = DataTransfer.createDataTransfer();
+    DataTransferContent.setHtmlData(dataTransfer, content);
+    InputEvents.fireBeforeInputEvent(editor, 'insertFromPaste', { dataTransfer });
     InputEvents.fireInputEvent(editor, 'insertFromPaste');
```

Comment:
isn't the `beforeInput` event normally cancellable, which prevents the `input` event?

Reply:
Nice observation, resolved in https://github.com/tinymce/tinymce/pull/8769